packages/eslint-plugin-stories/src/rules/csf-v3-types.ts

```diff
@@ -0,0 +1,126 @@
+import type { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import isStories from '../utils/isStories';
+
+/**
+ * Enforce specifying an explicit type for story objects where necessary for `composeStories` and
+ * Storybook to correctly infer their args.
+ *
+ * For example, instead of
+ *
+ *   const Primary = {
+ *     render: () => { ... },
+ *   }
+ *
+ * this rules enforces adding a type annotation
+ *
+ *   const Primary: StoryObj<Args> = {
+ *     render: () => { ... },
+ *   }
+ *
+ * Only detects 2 cases:
+ *
+ * 1. Stories that are empty objects - `const Primary = {}`
+ * 2. Stories with a render function that takes args - `render: (args) => { ... }`
+ *
+ * Other cases exist that need types, but they're harder to detect statically. These 2 are likely
+ * the most common.
+ */
+const rule: TSESLint.RuleModule<'specifyType'> = {
```

Comment:
Note the use of `TSESLint.RuleModule` instead of the standard `Rule.RuleModule`.

Necessary to get god `@typescript-eslint/parser` types. They're compatible, but the `TSESLint` version enforces via the type system defining the rule in an oddly specific way.

For example, it forces us to define `meta`, and to specify failure messages via `messageId` (instead of directly passing a string to `report`).

TL;DR - all this is valid eslint rule config, but it still looks a bit different than other rules in this plugin.